R/finalize_plot.R

```diff
@@ -315,29 +338,113 @@ finalize_plot <- function(input_plot = NULL,
                     col = cmapplot_globals$colors$blackish)
   )
 
+
+  # Function to create plot object with left aligned legend on top
+  buildChart <- function(p,l_margins,p_margins,l_indent,l_height,p_height) {
+
+    # Add left indent to plot margins
+    updated_l_margins <- c(l_margins[1:3],l_margins[4] + l_indent)
+
+    # Create a ggplot table of the updated plot element, including a left-aligned legend
+    gt <- ggplot2::ggplot_gtable(
+      ggplot2::ggplot_build(p + theme(legend.position = "left",
+                                      legend.justification = "left",
+                                      legend.direction = "horizontal",
+                                      plot.title = element_blank(),
+                                      plot.margin = grid::unit(p_margins,"bigpts"),
+                                      legend.margin = margin(t = updated_l_margins[1],
+                                                             r = updated_l_margins[2],
+                                                             b = updated_l_margins[3],
+                                                             l = updated_l_margins[4],
+                                                             "bigpts")
+                                      )
+                            )
+      )
+
+    # Extract the legend
+    l <- gtable::gtable_filter(gt, "guide-box")[[1]]
+
+    # Remove the legend from the plot
+    p <- p + ggplot2::theme(legend.position = "none")
+
+    # Create a buffer
+    buffer <- grid::rectGrob(gp = grid::gpar(col = "transparent", fill = "transparent"))
+
+    # Assemble
+    built <- gridExtra::grid.arrange(
+      gridExtra::arrangeGrob(
```

Comment:
I believe that `grid.arrange` and `arrangeGrob` are the same thing, except that grid.arrange draws the figure and arrangeGrob just assembles it. Are you sure this doesn't work equally well without `grid.arrange()`?

Reply:
Yes I noticed that and addressed it already.